### PR TITLE
[Feature] added archive deb build

### PR DESCRIPTION
**Pull request by yourbuddyconner**
- Added artifacts release for archive node
  - Docker
  - `.deb`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -59,7 +59,53 @@ jobs:
...
+                CODA_DOCKER: true
+                HASURA_PORT: 8080
+            - image: postgres:12
+              environment:
+                POSTGRES_PASSWORD: codarules
+                POSTGRES_USER: admin
+                POSTGRES_DB: archiver
+            - image: hasura/graphql-engine:v1.0.0-beta.6
+              entrypoint: ["sh", "-c"]
+              command: ["sleep 20 && graphql-engine serve"]
```
Comment (wu-s-john) [inline .circleci/config.yml]: Is it possible to drop this to 5 seconds?

File: .circleci/config.yml [hunk 2]
```diff
@@ -59,7 +59,53 @@ jobs:
...
+            - run:
+                  name: Update OPAM
+                  command: ./scripts/update-opam-in-docker.sh
+            - run:
+                  name: Build Archive Process
+                  command:  bash -c 'eval `opam config env` && make build_archive'
+                  environment:
+                    DUNE_PROFILE: testnet_postake_medium_curves
+            - run:
+                  name: Set Up Database
```
Comment (wu-s-john) [inline .circleci/config.yml]: do we need the Postgres Docker image above?

File: .circleci/config.yml [hunk 3]
```diff
@@ -59,7 +59,53 @@ jobs:
...
+                  command: | 
+                    sudo apt-get install -y postgresql 
+                    PGPASSWORD=codarules psql -h localhost -p 5432 -U admin -d archiver -a -f src/app/archive/create_schema.sql
+            - run:
+                  name: Configure Hasura
+                  command: ls && ./scripts/archive/make_hasura_configurations.sh
+            #- run:
```
Comment (wu-s-john) [inline .circleci/config.yml]: remove comments

File: .circleci/config.yml [hunk 4]
```diff
@@ -59,7 +59,53 @@ jobs:
...
+                  environment:
+                    DUNE_PROFILE: testnet_postake_medium_curves
+            - run:
+                  name: Set Up Database
+                  command: | 
+                    sudo apt-get install -y postgresql 
+                    PGPASSWORD=codarules psql -h localhost -p 5432 -U admin -d archiver -a -f src/app/archive/create_schema.sql
+            - run:
+                  name: Configure Hasura
+                  command: ls && ./scripts/archive/make_hasura_configurations.sh
```
Comment (wu-s-john) [inline .circleci/config.yml]: I don't think you need the `ls` command.

File: .circleci/config.yml (hunk 1 above)
Comment (yourbuddyconner) [reply .circleci/config.yml]: It doesn't affect the ultimate build-time, I found 10 seconds was sufficient but doubled that out of an abundance of caution.

File: .circleci/config.yml (hunk 2 above)
Comment (yourbuddyconner) [reply .circleci/config.yml]: Yes, the DB image is required, in addition to this step. If we were using a machine executor we would be able to mount the sql file in as a volume, this is the workaround.

File: .circleci/config.yml (hunk 3 above)
Comment (yourbuddyconner) [reply .circleci/config.yml]: @wu-s-john Do we even need the make/check configurations steps?

File: .circleci/config.yml (hunk 2 above)
Comment (wu-s-john) [reply .circleci/config.yml]: ah okay. Can you add that as a comment? It was confusing why you had this step.

File: .circleci/config.yml (hunk 1 above)
Comment (wu-s-john) [reply .circleci/config.yml]: ah let's try 10 seconds for now and if it fails again, then we will change it 15 seconds and then to 20 seconds.

File: .circleci/config.yml (hunk 3 above)
Comment (wu-s-john) [reply .circleci/config.yml]: We definitely need the make configuration step. The make step sends metadata to Hasura that configures information for Hasura, i.e. reference columns, encapsulating some columns for querying/mutations.

**Pull request merged**